Source/WebCore/rendering/BackgroundPainter.cpp

```diff
@@ -134,6 +134,8 @@ void BackgroundPainter::paintFillLayers(const Color& color, const FillLayer& fil
 
     auto& context = m_paintInfo.context();
     auto baseBgColorUsage = BaseBackgroundColorUse;
+    if (m_paintInfo.paintBehavior & PaintBehavior::AllowTransparentBackground)
```

Comment:
> Which renderers is this consulted on
what does this mean exactly?

> Is it only the root background that we want to make opaque?
The intention is that for any part of the PDF where the background would be forced to white, to just leave it as transparent if it was originally transparent.

> Why are we not sharing the code we use for making the view background transparent?
I didn't know such code existed. Do you know where it is?

Reply:
Look around `FrameView::isTransparent()` and `RenderView::shouldPaintBaseBackground()`

Reply:
@smfr If I change those, wouldn't the web page itself change to become transparent? (instead of only having the transparency-ness apply to the generated PDF)